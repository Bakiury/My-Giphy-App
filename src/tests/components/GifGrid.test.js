import { shallow } from 'enzyme';
import React from 'react';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); // Simular hook

describe('Probando <GifGrid />', () => {

    const category = 'Goku';
    // const wrapper = shallow(<GifGrid category={category} />);

    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: []
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/abc/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs
        });

        const wrapper = shallow(<GifGrid category={category} />);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});