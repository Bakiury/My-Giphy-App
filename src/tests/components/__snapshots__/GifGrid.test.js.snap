// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifGrid /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
